docs/docs/context.md

```diff
@@ -198,14 +202,6 @@ MediaQuery.childContextTypes = {
 };
 ```
 
-## When not to use context
-
-Just as global variables are best avoided when writing clear code, you should avoid using context in most cases. In particular, think twice before using it to "save typing" and using it instead of passing explicit props.
-
-The best use cases for context are for implicitly passing down the logged-in user, the current language, or theme information. All of these might otherwise be true globals, but context lets you scope them to a single React subtree.
-
-Do not use context to pass your model data through components. Threading your data through the tree explicitly is much easier to understand. Using context makes your components more coupled and less reusable, because they behave differently depending on where they're rendered.
-
-## Known limitations
+## Known Limitations
 
 If a context value provided by a component changes, descendants that use that value won't update if an intermediate parent returns `false` from `shouldComponentUpdate`. See issue [#2517](https://github.com/facebook/react/issues/2517) for more details.
```

Comment:
OK, I added a bunch more about "why not to use context". I also just added a link to that article.